redis/asyncio/client.py

```diff
@@ -511,12 +511,17 @@ async def execute_command(self, *args, **options):
         if self.single_connection_client:
             await self._single_conn_lock.acquire()
         try:
-            return await conn.retry.call_with_retry(
-                lambda: self._send_command_parse_response(
-                    conn, command_name, *args, **options
-                ),
-                lambda error: self._disconnect_raise(conn, error),
+            return await asyncio.shield(
```

Comment:
The try...finally should be inside the asyncio.shield if you want to prevent disconnects.